### PR TITLE
Create GTK+ Implementation for Switch widget

### PR DESCRIPTION
**Pull request by DariusMontez**
Current features:

 - a label, aligned to the left of the native switch
 - `on_toggle` handler
 - set initial state with `Switch(is_on=True)`

Don't mind the changes to Table and Button, they have already been merged in a previous PR, but I somehow committed them again. I'm not exactly sure how that happened.

### CODE REVIEW

**Comment by DariusMontez** [comment]
Oh. It looks like GitHub figured it all out. Yay Github!

**Comment by Ocupe** [comment]
@DariusMontez great to see more from you! Would you mind creating a little `toga/example/switch` app that shows the functionality of the switch implementation? You can use mine as a starting point [LINK](https://github.com/Ocupe/toga_test_app_collection) if you like. This way not every developer has to write their own test app and users can use it as a example.

**Comment by DariusMontez** [comment]
While writing a demo app, I found that simply changing the switch label text causes the widget to immediately move to the top left. I am not yet familiar with Toga's layout system, so I'm not sure what is going on. I'll go ahead and commit what I've got, since the demo app itself should be fine.

Running this little snippet should reproduce my issue:

```python
import toga

def on_toggle(switch):
        switch.label = "toggled!"
        
switch = toga.Switch("Toggle me -->", on_toggle=on_toggle)
box = toga.Box(children=[switch])
box.style.set(padding=32)
toga.App("Switches", "app.id", startup=lambda app: box).main_loop()
```

**Comment by DariusMontez** [comment]
@Ocupe I just found issue #270 ! Since this is a known issue, it means my Switch widget is probably fine. It still would be great to figure out what is going on though.

**Comment by Ocupe** [comment]
@DariusMontez thank you for the nice little example app 👍.

Short note on the changes on `Table` and `Button`. I don't see that the changes have been merged before. Have you updated your branch? You should probably open a new PR for changes that are not related to this PR. If you need help updating your branch hit me up on Gitter.

Issue #270 is probably my fault but if you would like to take a stab I don't mind :P

Regrading the widget test collection. I would like to see them merged but I also know that some of the apps are low quality. So every time I, or someone else, works on a widget I try to get the corresponding demo app merged. This way we hopefully soon have a little collection of high quality example apps for all the widgets.

**Comment by DariusMontez** [comment]
Sorry for the confusion on `Table` and `Button` - before I created this PR, I updated my local fork from upstream, and it (Github) showed several that several changes were made to `Button` and `Table`, but I recognized them as the already merged changes from #268 . This is what I was trying to describe in my first comment.

Sometime later, I unintentionally committed 0b66e8f. I'll revert when I'm able. I've been playing with different ways to implement handlers, so even if that commit is helpful, I'd like to have some discussion to establish a design pattern for event handling, both internally and externally. 

I don't know much about Gtk's layout allocation system, but I did find from the docs that `Gtk.Label` will call `GtkWidget::queue_resize()` if you change the label text, so the problem might be traceable from that point forward.

File: examples/switch/setup.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    long_description = readme.read()
+
+
+setup(
+    name='switch',
+    version=version,
+    description='A demonstration of all features of a toga.Switch example for the native GUI toolkit, Toga.',
+    long_description=long_description,
+    author='DariusMontez',
+    author_email='darius.montez@gmail.com',
```
Comment (phildini) [inline examples/switch/setup.py]: I think we're preferring 'BeeWare' and 'contact@pybee.org' for these entries, but this is more of a nit than a blocker.

**Comment by DariusMontez** [comment]
I got this PR cleaned up a bit. I'm ready for review!

File: examples/switch/setup.py (hunk 1 above)
Comment (DariusMontez) [reply examples/switch/setup.py]: Not a problem. Thank's for letting me know.

**Pull request merged**